File: Python/happy-number.py
class Solution:
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        s = set()

        while n != 1:
            if n in s:
                return False
            s.add(n)

            temp = 0
            while n > 0:
                temp += pow(n % 10, 2)
                n //= 10
            n = temp

        return True

File: Swift/happy-number.swift
class Solution {
    func isHappy(_ n: Int) -> Bool {
        var set = Set<Int>()
        var n = n

        while n != 1 {
            if set.contains(n) {
                return false
            }
            set.insert(n)

            var temp = 0
            while n > 0 {
                temp += (n % 10) * (n % 10)
                n /= 10
            }
            n = temp
        }

        return true
    }
}
